### PR TITLE
Refactored the Recode Numeric dialog

### LINKED ISSUE #6472: Refactor the Recode Numeric dialog so that it is implemented in the standard way
In PR #6434, @Patowhiz  wrote:

>For some reasons dlgRecodeNumeric is implemented differently, for instance the setting of R code is not done in the 'normal' way and parameters are also manually written. So I only focused on adding the save control. However, I think it should be redone in 'normal' way because I see no reason as to why it shouldn't

Please see additional comments in PR #6434.

### PR DESCRIPTION
**Pull request by Ivanluv**
fixes #6472
@africanmathsinitiative/developers  this is ready for review

### CODE REVIEW

**Review by rdstern — CHANGES_REQUESTED** [review]
Has this been tested before.  I get the "usual" developer error when going into the dialogue - each time.
Apart from that, it seems to work ok.
It is a good example for an intern to investigate and possibly improve further.  It isn't an easy dialogue to use - and it is useful!
So, once the developer error is sorted, then could this be merged, and a new issue be made.  This would be to evaluate improving the dialogue.  This could be just by adding tooltips, or by adding extra options, or by considering using the cut2 function from Hmisc instead.  
I am happy to write this new issue.

**Comment by Ivanluv** [comment]
@rdstern  the developer error is fixed

**Review by rdstern — APPROVED** [review]
Great

**Comment by lloyddewit** [comment]
@N-thony Please could you review? thanks

File: instat/dlgRecodeNumeric.vb [hunk 1]
```diff
@@ -148,51 +153,31 @@ Public Class dlgRecodeNumeric
...
-                ucrBase.clsRsyntax.AddParameter("right", "TRUE")
-            Else
-                ucrBase.clsRsyntax.RemoveParameter("right")
-            End If
+    Private Sub ucrChkAddLabels_ControlValueChanged(ucrChangedControl As ucrCore) Handles ucrChkAddLabels.ControlValueChanged
+        If ucrChkAddLabels.Checked Then
+            AddLabelsParameter()
         Else
-            ucrBase.clsRsyntax.AddParameter("right", "FALSE")
+            clsCutFunction.RemoveParameterByName("labels")
```
Comment (N-thony) [inline instat/dlgRecodeNumeric.vb:180]: This parameter is removed here but there is no place that is added.

**Review by N-thony — CHANGES_REQUESTED** [review]

File: instat/dlgRecodeNumeric.vb (hunk 1 above)
Comment (Ivanluv) [reply instat/dlgRecodeNumeric.vb:180]: it's is done on line 147

File: instat/dlgRecodeNumeric.vb [hunk 2]
```diff
@@ -85,13 +78,37 @@ Public Class dlgRecodeNumeric
...
 
         If Not ucrReceiverRecode.IsEmpty() AndAlso Not ucrMultipleNumericRecode.IsEmpty AndAlso ucrSaveRecode.IsComplete Then
-            If chkAddLabels.Checked AndAlso Not ucrMultipleLabels.IsEmpty Then
-                If (ucrMultipleNumericRecode.clsRList.clsParameters.Count > 1 AndAlso (Not ucrMultipleLabels.clsRList.clsParameters.Count <> ucrMultipleNumericRecode.clsRList.clsParameters.Count - 1)) OrElse (ucrMultipleNumericRecode.clsRList.clsParameters.Count = 1 AndAlso Integer.TryParse(ucrMultipleNumericRecode.clsRList.clsParameters(0).strArgumentValue, iTemp) AndAlso iTemp = ucrMultipleLabels.clsRList.clsParameters.Count) Then
+            If ucrChkAddLabels.Checked AndAlso Not ucrInputMultipleLabels.IsEmpty Then
+                If (ucrMultipleNumericRecode.clsRList.clsParameters.Count > 1 AndAlso (Not ucrInputMultipleLabels.clsRList.clsParameters.Count <> ucrMultipleNumericRecode.clsRList.clsParameters.Count - 1)) OrElse (ucrMultipleNumericRecode.clsRList.clsParameters.Count = 1 AndAlso Integer.TryParse(ucrMultipleNumericRecode.clsRList.clsParameters(0).strArgumentValue, iTemp) AndAlso iTemp = ucrInputMultipleLabels.clsRList.clsParameters.Count) Then
```
Comment (N-thony) [inline instat/dlgRecodeNumeric.vb:111]: If (ucrMultipleNumericRecode.clsRList.clsParameters.Count > 1 AndAlso (Not ucrInputMultipleLabels.clsRList.clsParameters.Count <> ucrMultipleNumericRecode.clsRList.clsParameters.Count - 1)) OrElse (ucrMultipleNumericRecode.clsRList.clsParameters.Count = 1 AndAlso Integer.TryParse(ucrMultipleNumericRecode.clsRList.clsParameters(0).strArgumentValue, Nothing) AndAlso isNothing(ucrInputMultipleLabels.clsRList.clsParameters.Count)) Then

and no need to declare iTemp?

Comment (Ivanluv) [reply instat/dlgRecodeNumeric.vb:111]: @N-thony  that's used to parse  each  `Rparameter's ``strArgumentValue `  
 in `ucrMultipleNumericRecode.clsRList `to integers

Comment (N-thony) [reply instat/dlgRecodeNumeric.vb:111]: That is also good. Over to you @lloyddewit.

File: instat/dlgRecodeNumeric.vb [hunk 3]
```diff
@@ -85,13 +78,37 @@ Public Class dlgRecodeNumeric
         strDefaultDataFrame = ""
         strDefaultColumn = ""
     End Sub
+    Private Sub SetDefaults()
```
Comment (lloyddewit) [inline instat/dlgRecodeNumeric.vb:81]:
```suggestion

    Private Sub SetDefaults()
```

**Review by lloyddewit — CHANGES_REQUESTED** [review]

**Review by lloyddewit — DISMISSED** [review]

**Comment by lloyddewit** [comment]
@rdstern Please could you retest and approve? thanks

**Review by rdstern — DISMISSED** [review]

**Review by lloyddewit — APPROVED** [review]

**Comment by lloyddewit** [comment]
@rdstern I had to resolve a merge conflict in this PR. Please could you approve again so we can merge? Thanks

**Review by rdstern — APPROVED** [review]

**Pull request merged**